crypto/dh/dh_check.c

```diff
@@ -257,9 +257,27 @@ int ossl_dh_check_priv_key(const DH *dh, const BIGNUM *priv_key, int *ret)
     two_powN = BN_new();
     if (two_powN == NULL)
         return 0;
-    if (dh->params.q == NULL)
+
+    if (dh->params.q != NULL) {
+        upper = dh->params.q;
+#ifndef FIPS_MODULE
+    /*
+     * We do not have q so we just check the key is within the range
+     * we generate it.
+     */
+    } else if (dh->params.p != NULL) {
+        int32_t length = dh->length;
+
+        if (length == 0)
+            length = BN_num_bits(dh->params.p) - 1;
+        if (BN_num_bits(priv_key) <= length
+            && BN_num_bits(priv_key) > 1)
+            ok = 1;
+        goto err;
```

Comment:
This seems to be missing a check that bit 0 is 0 in case of g = 2, p % 8 == 3

Reply:
No, there is no point in such check. Where in PKCS#3 do you see such requirement.